cmd/installer/pkg/install/install.go

```diff
@@ -403,8 +410,8 @@ func (i *Installer) Install(ctx context.Context, mode Mode) (err error) {
 		var ok bool
 
 		if mode == ModeUpgrade {
-			if ok, err = metaState.SetTag(context.Background(), meta.Upgrade, i.bootloader.PreviousLabel()); !ok || err != nil {
-				return fmt.Errorf("failed to set upgrade tag: %q", i.bootloader.PreviousLabel())
+			if ok, err = metaState.SetTag(context.Background(), meta.Upgrade, bootInstallResult.PreviousLabel); !ok || err != nil {
```

Comment:
`Install` method has `ctx` as the param, but we use `context.Background()` here. I wonder if we can change that to be `ctx`.

Reply:
I'll fix it